pkg/backend/stack.go

```diff
@@ -171,7 +172,8 @@ func GetEnvironmentTagsForCurrentStack() (map[apitype.StackTagName]string, error
 	// Tags based on Pulumi.yaml.
 	projPath, err := workspace.DetectProjectPath()
 	if err != nil {
-		return nil, err
```

Comment:
Maybe wrap this function in `GetEnvironmentTagsForStack(stackRef StackReference)` then fallback to `GetEnvironmentTagsForCurrentStack()`